src/components/player/player.js

```diff
@@ -1,11 +1,77 @@
-import React from "react"
+import React, { useEffect, useState } from "react"
 import { PlayButton } from "react-soundplayer/components"
 import { withCustomAudio } from "react-soundplayer/addons"
 
+import fetchJsonp from "fetch-jsonp"
+
 const Player = withCustomAudio(props => {
+  const [streamData, setStreamData] = useState({ streamstatus: 1 })
+  const [mixcloudData, setMixcloudData] = useState("")
+
+  const fetchMixcloudStream = url => {
+    if (!url) url = "https://api.mixcloud.com/entropyfm/feed/"
+    fetch(url)
+      .then(resp => resp.json())
+      .then(data => {
+        for (let i = 0; i < 10; i++) {
+          // 10 tries to fetch a random mixcloud recording
+          let temp = data.data[Math.floor(Math.random() * data.data.length)]
+          if (temp.cloudcasts) {
+            setMixcloudData(temp.cloudcasts[0].key)
+            return
+          }
+        }
+        // No mixcloud recording found, go to next page
+        if (data.paging.next) fetchMixcloudStream(data.paging.next)
+      })
```

Comment:
I'm confused by this logic. I thought the goal was to pull the most recent show, not a random one?